ckan/lib/plugins.py

```diff
@@ -70,7 +73,14 @@ def lookup_group_controller(group_type=None):
     Returns the group controller associated with the given group type. The
     controller is expressed as a string that you'd pass to url_to(controller=x)
     """
-    return _group_controllers.get(group_type)
+    return _group_controllers.get(group_type)   
+
+
+def lookup_group_blueprints(group_type=None):
```

Comment:
I don't think we'll need different blueprints. The mechanism should be the same as we have now. Different `IGroupForm` plugins get registered, and in `register_group_plugins()` we register the custom URLs (eg `/theme`) pointing the correct blueprint (`group` or `organization`). Basically we need to replace calls to the Pylons routes mapper with calls to the `.add_url_rule()` of the relevant blueprint.

The challenge might be to import the blueprint instances avoiding circular redirects. We might need to do it just before registering the routes.

```python
            map.connect('%s_index' % group_type, '/%s' % group_type, controller=group_controller, action='index')
```

Reply:
Actually ignore me. We need different blueprints because we have to change the `url_prefix` of each.

So perhaps the best is to abstract the registering of the routes on the blueprint to a function and call that when registering the plugins. Something like

```python

# views/group.py

def add_group_url_rules(blueprint):
    blueprint.add_url_rule('u'/', view_func=index, strict_slashes=False)
    # ...

def add_org_url_rules(blueprint):
   #  Do we need two functions? Have groups and org different routes?
   blueprint.add_url_rule('u'/', view_func=index, strict_slashes=False)

group = Blueprint('group', __name__, url_prefix=u'/group')
organization = Blueprint('organization', __name__, url_prefix=u'/organization')

add_group_url_rules(group)
add_org_url_rules(organization)


# lib/plugins.py

def register_group_plugins():
     # ...
     for group_types in plugin.group_types():
        custom_type_blueprint = Blueprint(group_type, __name__, url_prefix=u'/' + group_type)
        add_group_url_rules(custom_type_blueprint)

```

WDYT @smotornyuk? For datasets we will need something similar